extensions/amp-analytics/0.1/requests.js

```diff
@@ -258,15 +268,26 @@ export class RequestHandler {
    */
   constructExtraUrlParamStrs_(baseUrl, extraUrlParamStrsPromise) {
     return Promise.all(extraUrlParamStrsPromise).then(paramStrs => {
-      filterSplice(paramStrs, item => {return !!item;});
-      const extraUrlParamsStr = paramStrs.join('&');
+      const extraUrlParamStrs = {
+        requestUrl: baseUrl,
+        requestBody: '',
+      };
       let requestUrl;
-      if (baseUrl.indexOf('${extraUrlParams}') >= 0) {
-        requestUrl = baseUrl.replace('${extraUrlParams}', extraUrlParamsStr);
+      if (!this.useBody_) {
+        filterSplice(paramStrs, item => {return !!item;});
+        const extraUrlParams = paramStrs.join('&');
+        if (baseUrl.indexOf('${extraUrlParams}') >= 0) {
+          requestUrl = baseUrl.replace('${extraUrlParams}', extraUrlParams);
```

Comment:
I believe this is super rare case, but if some one request `${extraUrlParams}` in baseUrl, we should replace it even with `useBody: true`.

Reply:
me and @oliverfernandez talked about this offline while I planning the solution and in the end we decided not to support this as it overcomplicates the code because it means we have to expand the params twice, once as queryParamString and once as JSON just to support this. For us having both useBody and extraUrlParams in the requestUrl should be a configuration error, because you are duplicating the request payload. I can change it and throw an error if this is the case. WDYT?

Reply:
Throwing an error before there's a request sounds good. Make sure you add a comment to explain the reason. Thanks